In [1]:
import pylab as pl
import pandas as pd
import geopandas as gp
import numpy as np
import scipy.stats
import os
import json
from pandas.tools.plotting import scatter_matrix
import statsmodels.api as sm
#lowercase ols
import statsmodels.formula.api as smf
#uppercase OLS
import datetime as dt
import re


%pylab inline

Populating the interactive namespace from numpy and matplotlib


# 1. Define Research Question

### Which Community Districts in NYC show the highest number of complaints?

# 2. Think about what data are available
 - Acquiured data from CUSP data facility for 311 complaints from 2010 to present

In [2]:
data = pd.read_csv('/projects/open/NYCOpenData/nycopendata/data/erm2-nwe9/1446832678/erm2-nwe9')

/opt/rh/anaconda/root/envs/PUI2016_Python3/lib/python3.5/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (8,17,39,40,47,48) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
data.head()

Unique Key            Created Date             Closed Date Agency  \
0    31911011  11/05/2015 02:59:15 AM                     NaN    DOT   
1    31908754  11/05/2015 02:09:49 AM                     NaN  CHALL   
2    31910423  11/05/2015 02:06:51 AM                     NaN    DPR   
3    31909924  11/05/2015 02:02:20 AM                     NaN   NYPD   
4    31913310  11/05/2015 01:57:20 AM  11/05/2015 01:57:31 AM    HRA   

                          Agency Name                 Complaint Type  \
0        Department of Transportation               Street Condition   
1                               CHALL          Opinion for the Mayor   
2  Department of Parks and Recreation  Root/Sewer/Sidewalk Condition   
3     New York City Police Department                Illegal Parking   
4        HRA Benefit Card Replacement       Benefit Card Replacement   

                    Descriptor       Location Type Incident Zip  \
0                      Pothole                 NaN        11224   
1                      HOUSING                 NaN          NaN   
2  Trees and Sidewalks Program              Street        11234   
3              Blocked Hydrant     Street/Sidewalk        11218   
4                     Medicaid  NYC Street Address          NaN   

      Incident Address                    ...                     \
0                  NaN                    ...                      
1                  NaN                    ...                      
2  1157 EAST 57 STREET                    ...                      
3    722 EAST 4 STREET                    ...                      
4                  NaN                    ...                      

  Bridge Highway Name Bridge Highway Direction Road Ramp  \
0                 NaN                      NaN       NaN   
1                 NaN                      NaN       NaN   
2                 NaN                      NaN       NaN   
3                 NaN                      NaN       NaN   
4                 NaN                      NaN       NaN   

  Bridge Highway Segment Garage Lot Name Ferry Direction Ferry Terminal Name  \
0                    NaN             NaN             NaN                 NaN   
1         1-1-1173130914             NaN             NaN                 NaN   
2                    NaN             NaN             NaN                 NaN   
3                    NaN             NaN             NaN                 NaN   
4                    NaN             NaN             NaN                 NaN   

    Latitude  Longitude                                  Location  
0  40.573431 -73.991742   (40.57343122248129, -73.99174247588253)  
1        NaN        NaN                                       NaN  
2  40.625004 -73.920726   (40.62500363580505, -73.92072558378698)  
3  40.634522 -73.974790  (40.634522428879706, -73.97479041437481)  
4        NaN        NaN                                       NaN  

[5 rows x 53 columns]

In [4]:
#data.columns

In [5]:
comdistcalls = data[['Unique Key', 'Created Date', 'Closed Date', 'Agency', 'Agency Name',
       'Complaint Type', 'Descriptor','Incident Zip',
       'City', 'Landmark', 'Facility Type', 'Status', 'Due Date',
       'Resolution Description', 'Resolution Action Updated Date',
       'Community Board', 'Borough','Latitude', 'Longitude', 'Location']]

In [6]:
comdistcalls.head()

Unique Key            Created Date             Closed Date Agency  \
0    31911011  11/05/2015 02:59:15 AM                     NaN    DOT   
1    31908754  11/05/2015 02:09:49 AM                     NaN  CHALL   
2    31910423  11/05/2015 02:06:51 AM                     NaN    DPR   
3    31909924  11/05/2015 02:02:20 AM                     NaN   NYPD   
4    31913310  11/05/2015 01:57:20 AM  11/05/2015 01:57:31 AM    HRA   

                          Agency Name                 Complaint Type  \
0        Department of Transportation               Street Condition   
1                               CHALL          Opinion for the Mayor   
2  Department of Parks and Recreation  Root/Sewer/Sidewalk Condition   
3     New York City Police Department                Illegal Parking   
4        HRA Benefit Card Replacement       Benefit Card Replacement   

                    Descriptor Incident Zip      City Landmark Facility Type  \
0                      Pothole        11224  BROOKLYN      NaN           NaN   
1                      HOUSING          NaN       NaN      NaN           NaN   
2  Trees and Sidewalks Program        11234  BROOKLYN      NaN           NaN   
3              Blocked Hydrant        11218  BROOKLYN      NaN      Precinct   
4                     Medicaid          NaN       NaN      NaN           NaN   

       Status                Due Date  \
0        Open                     NaN   
1  Email Sent  11/19/2015 02:10:47 AM   
2        Open  12/05/2015 02:06:51 AM   
3    Assigned  11/05/2015 10:02:20 AM   
4      Closed                     NaN   

                              Resolution Description  \
0                                                NaN   
1  Your comments have been submitted to the Mayor...   
2  The Department of Parks and Recreation usually...   
3  Your complaint has been received by the Police...   
4  The Human Resources Administration received yo...   

  Resolution Action Updated Date Community Board      Borough   Latitude  \
0         11/05/2015 02:59:15 AM     13 BROOKLYN     BROOKLYN  40.573431   
1                            NaN   0 Unspecified  Unspecified        NaN   
2                            NaN     18 BROOKLYN     BROOKLYN  40.625004   
3         11/05/2015 02:07:18 AM     12 BROOKLYN     BROOKLYN  40.634522   
4                            NaN   0 Unspecified  Unspecified        NaN   

   Longitude                                  Location  
0 -73.991742   (40.57343122248129, -73.99174247588253)  
1        NaN                                       NaN  
2 -73.920726   (40.62500363580505, -73.92072558378698)  
3 -73.974790  (40.634522428879706, -73.97479041437481)  
4        NaN                                       NaN

In [7]:
#checking for all Boroughs in data

comdistcalls['Borough'].unique()

array(['BROOKLYN', 'Unspecified', 'STATEN ISLAND', 'MANHATTAN', 'BRONX',
       'QUEENS'], dtype=object)

In [8]:
# Checking for all Community Board Districts

comdistcalls['Community Board'].unique()

array(['13 BROOKLYN', '0 Unspecified', '18 BROOKLYN', '12 BROOKLYN',
       '02 STATEN ISLAND', '08 MANHATTAN', '06 BROOKLYN', '10 BRONX',
       '06 MANHATTAN', '04 BROOKLYN', '04 QUEENS', '09 MANHATTAN',
       '14 BROOKLYN', '05 MANHATTAN', '02 MANHATTAN', '10 QUEENS',
       '07 QUEENS', '02 QUEENS', '11 BROOKLYN', '05 QUEENS', '05 BROOKLYN',
       '09 QUEENS', '12 MANHATTAN', '01 BROOKLYN', '11 MANHATTAN',
       '03 BRONX', '03 STATEN ISLAND', '08 BROOKLYN', '03 BROOKLYN',
       '07 MANHATTAN', '10 MANHATTAN', '04 BRONX', '01 MANHATTAN',
       '07 BRONX', '09 BRONX', '03 MANHATTAN', '13 QUEENS', '02 BRONX',
       '12 QUEENS', '16 BROOKLYN', '08 QUEENS', '04 MANHATTAN',
       '14 QUEENS', '10 BROOKLYN', '11 QUEENS', '06 BRONX', '08 BRONX',
       '05 BRONX', '15 BROOKLYN', '02 BROOKLYN', '12 BRONX', '03 QUEENS',
       '06 QUEENS', '01 STATEN ISLAND', '01 BRONX', '01 QUEENS',
       '17 BROOKLYN', '11 BRONX', 'Unspecified QUEENS', '09 BROOKLYN',
       'Unspecified MANHATTAN'

In [9]:
#length of unniqe values
len(comdistcalls['Community Board'].unique())

# There are 77 because of the "Unspecified" borough districts and non-real community board. 

77

In [10]:
#Check for dupicates, must be by unique identifier

comdistcalls['dupcheck'] = comdistcalls.duplicated('Unique Key')

/opt/rh/anaconda/root/envs/PUI2016_Python3/lib/python3.5/site-packages/ipykernel/__main__.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  app.launch_new_instance()


In [11]:
comdistcalls.groupby(['dupcheck']).count()

Unique Key  Created Date  Closed Date    Agency  Agency Name  \
dupcheck                                                                 
False       10187744      10187744      9736865  10187744     10187744   
True              22            22           22        22           22   

          Complaint Type  Descriptor  Incident Zip     City  Landmark  \
dupcheck                                                                
False           10187744    10156687       9393585  9399032      7310   
True                  22          22            22       22         0   

          Facility Type    Status  Due Date  Resolution Description  \
dupcheck                                                              
False           1608281  10187744   3067458                 6401005   
True                  0        22        22                      10   

          Resolution Action Updated Date  Community Board   Borough  Latitude  \
dupcheck                                                                        
False                            9936812         10187744  10187744   9308992   
True                                  20               22        22        22   

          Longitude  Location  
dupcheck                       
False       9308992   9308992  
True             22        22

In [12]:
#there are only 22 duplicates,  drop
comdistcalls.drop_duplicates('Unique Key')

Unique Key            Created Date             Closed Date Agency  \
0           31911011  11/05/2015 02:59:15 AM                     NaN    DOT   
1           31908754  11/05/2015 02:09:49 AM                     NaN  CHALL   
2           31910423  11/05/2015 02:06:51 AM                     NaN    DPR   
3           31909924  11/05/2015 02:02:20 AM                     NaN   NYPD   
4           31913310  11/05/2015 01:57:20 AM  11/05/2015 01:57:31 AM    HRA   
5           31914148  11/05/2015 01:55:55 AM                     NaN    DOT   
6           31912764  11/05/2015 01:55:37 AM                     NaN    DPR   
7           31912611  11/05/2015 01:47:12 AM                     NaN   NYPD   
8           31913909  11/05/2015 01:39:31 AM  11/05/2015 02:35:44 AM   NYPD   
9           31914856  11/05/2015 01:38:35 AM  11/05/2015 02:33:33 AM   NYPD   
10          31911308  11/05/2015 01:37:14 AM                     NaN   NYPD   
11          31909392  11/05/2015 01:36:43 AM                     NaN    TLC   
12          31912284  11/05/2015 01:30:36 AM  11/05/2015 02:13:35 AM   NYPD   
13          31911299  11/05/2015 01:28:52 AM  11/05/2015 03:08:36 AM   NYPD   
14          31909030  11/05/2015 01:27:21 AM  11/05/2015 02:03:06 AM   NYPD   
15          31909884  11/05/2015 01:20:26 AM                     NaN   NYPD   
16          31913776  11/05/2015 01:14:58 AM  11/05/2015 02:50:37 AM   NYPD   
17          31911881  11/05/2015 01:13:34 AM                     NaN   NYPD   
18          31913779  11/05/2015 01:11:44 AM  11/05/2015 01:59:55 AM   NYPD   
19          31915178  11/05/2015 01:10:11 AM  11/05/2015 02:53:47 AM   NYPD   
20          31907956  11/05/2015 01:07:17 AM  11/05/2015 02:39:12 AM   NYPD   
21          31913892  11/05/2015 01:06:28 AM                     NaN    DPR   
22          31910149  11/05/2015 01:00:42 AM                     NaN   NYPD   
23          31910350  11/05/2015 12:59:14 AM  11/05/2015 02:27:18 AM   NYPD   
24          31913895  11/05/2015 12:58:57 AM                     NaN    TLC   
25          31913898  11/05/2015 12:58:13 AM  11/05/2015 01:30:23 AM   NYPD   
26          31910446  11/05/2015 12:57:00 AM                     NaN    DOT   
27          31912897  11/05/2015 12:56:49 AM                     NaN   NYPD   
28          31913287  11/05/2015 12:56:00 AM                     NaN    DOT   
29          31909985  11/05/2015 12:55:56 AM                     NaN   NYPD   
...              ...                     ...                     ...    ...   
10187736    15634832  01/01/2010 12:00:00 AM  01/06/2010 12:00:00 AM    HPD   
10187737    15635330  01/01/2010 12:00:00 AM  02/04/2010 12:00:00 AM    HPD   
10187738    15635779  01/01/2010 12:00:00 AM  01/07/2010 12:00:00 AM    HPD   
10187739    15634856  01/01/2010 12:00:00 AM  01/07/2010 12:00:00 AM    HPD   
10187740    15634981  01/01/2010 12:00:00 AM  01/13/2010 12:00:00 AM    HPD   
10187741    15635661  01/01/2010 12:00:00 AM  01/06/2010 12:00:00 AM    HPD   
10187742    15635448  01/01/2010 12:00:00 AM  01/06/2010 12:00:00 AM    HPD   
10187743    15635825  01/01/2010 12:00:00 AM  01/03/2010 12:00:00 AM    HPD   
10187744    15629728  01/01/2010 12:00:00 AM  01/12/2010 12:00:00 AM    HPD   
10187745    15635199  01/01/2010 12:00:00 AM  01/09/2010 12:00:00 AM    HPD   
10187746    15634868  01/01/2010 12:00:00 AM  01/10/2010 12:00:00 AM    HPD   
10187747    15635375  01/01/2010 12:00:00 AM  01/03/2010 12:00:00 AM    HPD   
10187748    15634924  01/01/2010 12:00:00 AM  01/05/2010 12:00:00 AM    HPD   
10187749    15634760  01/01/2010 12:00:00 AM  01/05/2010 12:00:00 AM    HPD   
10187750    15634893  01/01/2010 12:00:00 AM  01/07/2010 12:00:00 AM    HPD   
10187751    15634905  01/01/2010 12:00:00 AM  01/25/2010 12:00:00 AM    HPD   
10187752    15635491  01/01/2010 12:00:00 AM  01/03/2010 12:00:00 AM    HPD   
10187753    15635355  01/01/2010 12:00:00 AM  01/08/2010 12:00:00 AM    HPD   
10187754    15634850  01/01/2010 12:00:00 AM  01/05/2010 12:

In [13]:
len(comdistcalls)

10187766

In [14]:
duplicates = data[data.duplicated('Unique Key')]

In [15]:
#Checking for total NaN values

comdistcalls.isnull().sum()

Unique Key                               0
Created Date                             0
Closed Date                         450879
Agency                                   0
Agency Name                              0
Complaint Type                           0
Descriptor                           31057
Incident Zip                        794159
City                                788712
Landmark                          10180456
Facility Type                      8579485
Status                                   0
Due Date                           7120286
Resolution Description             3786751
Resolution Action Updated Date      250934
Community Board                          0
Borough                                  0
Latitude                            878752
Longitude                           878752
Location                            878752
dupcheck                                 0
dtype: int64

### Patterns in missing values
 - Many of these columns such as "Landmark" and "Facility Type" have many NaN values.  
 - Other fields such as "Unique Key" which is probably an auto-generated field has zero
 - This pattern may be due to REQUIRED and OPTIONAL fields when submitting a complaint

### Data Editing:
 - Replacing missing "Resolution Descriptor" with "Complaint Type" is possible, as Complaint Type is probably the most similar value. However, leaving the field blank in this case may not affect the analysis. 
 - below I will go ahead and change these values

In [16]:
comdistcalls["Resolution Description"].fillna(comdistcalls["Complaint Type"], inplace=True)


/opt/rh/anaconda/root/envs/PUI2016_Python3/lib/python3.5/site-packages/pandas/core/generic.py:3191: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)


In [17]:
comdistcalls.isnull().sum()

Unique Key                               0
Created Date                             0
Closed Date                         450879
Agency                                   0
Agency Name                              0
Complaint Type                           0
Descriptor                           31057
Incident Zip                        794159
City                                788712
Landmark                          10180456
Facility Type                      8579485
Status                                   0
Due Date                           7120286
Resolution Description                   0
Resolution Action Updated Date      250934
Community Board                          0
Borough                                  0
Latitude                            878752
Longitude                           878752
Location                            878752
dupcheck                                 0
dtype: int64

As shown in the above output, there are no longer NaNs in the Complaint Type columnm

In [18]:
#This function will mark the "Unspecified" CB's with 0
def markers(data):
    if re.search(r"Unspecified", data):
        return 0
    elif int(re.search(r"(\d\d*)\s\w+", data).group(1))>18:
        return 0
    else:
        return 1

In [19]:
comdistcalls2  = comdistcalls["Community Board"].apply(markers)
#applying the function in a new 1-D array

In [20]:
comdistcalls["comdistcalls2"] = comdistcalls2

#adding a column to the original DF with the vlaues from the function

/opt/rh/anaconda/root/envs/PUI2016_Python3/lib/python3.5/site-packages/ipykernel/__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':


In [21]:
comdistcalls.head()

Unique Key            Created Date             Closed Date Agency  \
0    31911011  11/05/2015 02:59:15 AM                     NaN    DOT   
1    31908754  11/05/2015 02:09:49 AM                     NaN  CHALL   
2    31910423  11/05/2015 02:06:51 AM                     NaN    DPR   
3    31909924  11/05/2015 02:02:20 AM                     NaN   NYPD   
4    31913310  11/05/2015 01:57:20 AM  11/05/2015 01:57:31 AM    HRA   

                          Agency Name                 Complaint Type  \
0        Department of Transportation               Street Condition   
1                               CHALL          Opinion for the Mayor   
2  Department of Parks and Recreation  Root/Sewer/Sidewalk Condition   
3     New York City Police Department                Illegal Parking   
4        HRA Benefit Card Replacement       Benefit Card Replacement   

                    Descriptor Incident Zip      City Landmark      ...       \
0                      Pothole        11224  BROOKLYN      NaN      ...        
1                      HOUSING          NaN       NaN      NaN      ...        
2  Trees and Sidewalks Program        11234  BROOKLYN      NaN      ...        
3              Blocked Hydrant        11218  BROOKLYN      NaN      ...        
4                     Medicaid          NaN       NaN      NaN      ...        

                 Due Date                             Resolution Description  \
0                     NaN                                   Street Condition   
1  11/19/2015 02:10:47 AM  Your comments have been submitted to the Mayor...   
2  12/05/2015 02:06:51 AM  The Department of Parks and Recreation usually...   
3  11/05/2015 10:02:20 AM  Your complaint has been received by the Police...   
4                     NaN  The Human Resources Administration received yo...   

  Resolution Action Updated Date Community Board      Borough   Latitude  \
0         11/05/2015 02:59:15 AM     13 BROOKLYN     BROOKLYN  40.573431   
1                            NaN   0 Unspecified  Unspecified        NaN   
2                            NaN     18 BROOKLYN     BROOKLYN  40.625004   
3         11/05/2015 02:07:18 AM     12 BROOKLYN     BROOKLYN  40.634522   
4                            NaN   0 Unspecified  Unspecified        NaN   

   Longitude                                  Location  dupcheck comdistcalls2  
0 -73.991742   (40.57343122248129, -73.99174247588253)     False             1  
1        NaN                                       NaN     False             0  
2 -73.920726   (40.62500363580505, -73.92072558378698)     False             1  
3 -73.974790  (40.634522428879706, -73.97479041437481)     False             1  
4        NaN                                       NaN     False             0  

[5 rows x 22 columns]

In [22]:
comdistcallsREDUX = comdistcalls[comdistcalls.comdistcalls2 !=0]

In [23]:
#comdistcallsREDUX.head()

In [24]:
len(comdistcallsREDUX['Community Board'].unique())

59

In [25]:
pivot = comdistcallsREDUX["Unique Key"].groupby(comdistcallsREDUX["Community Board"]).count()

In [26]:
pivot.head(20)

Community Board
01 BRONX             74631
01 BROOKLYN         185057
01 MANHATTAN         77974
01 QUEENS           171484
01 STATEN ISLAND    182713
02 BRONX             60257
02 BROOKLYN         121022
02 MANHATTAN        133860
02 QUEENS           114333
02 STATEN ISLAND    121132
03 BRONX             75134
03 BROOKLYN         197306
03 MANHATTAN        150296
03 QUEENS           122009
03 STATEN ISLAND    136487
04 BRONX            181953
04 BROOKLYN         139706
04 MANHATTAN        132660
04 QUEENS            99316
05 BRONX            161078
Name: Unique Key, dtype: int64

In [27]:
# must create a data frame from the pivot table above

CBRank = pd.DataFrame(pivot)

In [28]:
CBRank.head()

Unique Key
Community Board             
01 BRONX               74631
01 BROOKLYN           185057
01 MANHATTAN           77974
01 QUEENS             171484
01 STATEN ISLAND      182713

In [29]:
CBRank["RANK"] = CBRank["Unique Key"].rank(ascending = False)

In [30]:
CBRank.head()

Unique Key  RANK
Community Board                   
01 BRONX               74631  57.0
01 BROOKLYN           185057   9.0
01 MANHATTAN           77974  55.0
01 QUEENS             171484  12.0
01 STATEN ISLAND      182713  10.0

In [31]:
CBRank = CBRank.reset_index()
#the column labels are off, must RESET index
#http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.reset_index.html

In [32]:
CBRank.head()

Community Board  Unique Key  RANK
0          01 BRONX       74631  57.0
1       01 BROOKLYN      185057   9.0
2      01 MANHATTAN       77974  55.0
3         01 QUEENS      171484  12.0
4  01 STATEN ISLAND      182713  10.0

In [40]:
#Renaming Column
CBRank.rename(columns={'Unique Key': 'Total Complaints'}, inplace=True)
CBRank.head(59)

Community Board  Total Complaints  RANK
0           01 BRONX             74631  57.0
1        01 BROOKLYN            185057   9.0
2       01 MANHATTAN             77974  55.0
3          01 QUEENS            171484  12.0
4   01 STATEN ISLAND            182713  10.0
5           02 BRONX             60257  59.0
6        02 BROOKLYN            121022  40.0
7       02 MANHATTAN            133860  32.0
8          02 QUEENS            114333  41.0
9   02 STATEN ISLAND            121132  39.0
10          03 BRONX             75134  56.0
11       03 BROOKLYN            197306   3.0
12      03 MANHATTAN            150296  21.0
13         03 QUEENS            122009  38.0
14  03 STATEN ISLAND            136487  30.0
15          04 BRONX            181953  11.0
16       04 BROOKLYN            139706  28.0
17      04 MANHATTAN            132660  33.0
18         04 QUEENS             99316  49.0
19          05 BRONX            161078  16.0
20       05 BROOKLYN            189765   6.0
21      05 MANHATTAN            142028  25.0
22         05 QUEENS            168570  13.0
23          06 BRONX             99547  48.0
24       06 BROOKLYN            114029  42.0
25      06 MANHATTAN            104249  45.0
26         06 QUEENS             90938  53.0
27          07 BRONX            185729   8.0
28       07 BROOKLYN            107031  44.0
29      07 MANHATTAN            151419  18.0
30         07 QUEENS            195961   4.0
31          08 BRONX             92733  52.0
32       08 BROOKLYN            141119  26.0
33      08 MANHATTAN            149076  22.0
34         08 QUEENS            123131  36.0
35          09 BRONX            147457  23.0
36       09 BROOKLYN            135833  31.0
37      09 MANHATTAN            127565  35.0
38         09 QUEENS            151242  19.0
39          10 BRONX             95496  50.0
40       10 BROOKLYN            111837  43.0
41      10 MANHATTAN            137344  29.0
42         10 QUEENS            128116  34.0
43          11 BRONX            122081  37.0
44       11 BROOKLYN            139890  27.0
45      11 MANHATTAN             95413  51.0
46         11 QUEENS            101908  47.0
47          12 BRONX            168455  14.0
48       12 BROOKLYN            150320  20.0
49      12 MANHATTAN            268053   1.0
50         12 QUEENS            229383   2.0
51       13 BROOKLYN             72362  58.0
52         13 QUEENS            163851  15.0
53       14 BROOKLYN            187794   7.0
54         14 QUEENS            103396  46.0
55       15 BROOKLYN            146129  24.0
56       16 BROOKLYN             85467  54.0
57       17 BROOKLYN            191720   5.0
58       18 BROOKLYN            156100  17.0

### As shown in the RANK Table above, Community Board 12 from Manhattan is ranked #1 with 268,053 complaints

# 4. Interpret Results